storage/isolation_manager.rs

```diff
@@ -360,61 +344,70 @@ impl Error for IsolationError {}
 #[derive(Debug)]
 struct Timeline {
     // We can adjust the Window size to amortise the cost of the read-write locks to maintain the timeline
-    next_relative_index_and_windows: RwLock<(i64, VecDeque<Arc<TimelineWindow<TIMELINE_WINDOW_SIZE>>>)>,
-    watermark: AtomicI64,
+    next_sequence_and_windows: RwLock<(SequenceNumber, VecDeque<Arc<TimelineWindow<TIMELINE_WINDOW_SIZE>>>)>,
```

Comment:
Let's move sequence umber back into Windows, and simplify all the related code `resolve_sequence_number` becomes `find_window(windows, sequence_number)` + `window.get_slot(sequence_number`, a more composable API.

Reply:
We also can just call this Windows and when we get the write lock to add more windows get the last window, and add a new one starting at LAST_WINDOW.sequence_number()+ WINDOW_SIZE